src/Features/Core/Portable/DocumentationCommentFormatting/AbstractDocumentationCommentFormattingService.cs

```diff
@@ -203,44 +182,36 @@ private static void AppendTextFromNode(FormatterState state, XNode node, Compila
             }
         }
 
-        private static void AppendTextFromTagWithNameAttribute(FormatterState state, XElement element, Compilation compilation)
+        private static void AppendTextFromAttribute(FormatterState state, XElement element, string attributeName)
         {
-            var nameAttribute = element.Attribute("name");
-
-            if (nameAttribute == null)
+            var attribute = element.Attribute(attributeName);
+            if (attribute == null)
             {
                 return;
             }
 
-            if (compilation != null && state.TryAppendSymbol(DocumentationCommentId.GetFirstSymbolForDeclarationId(nameAttribute.Value, compilation)))
-            {
-                return;
-            }
-            else
-            {
-                state.AppendString(TrimCrefPrefix(nameAttribute.Value));
-            }
+            state.AppendParts(CrefToSymbolDisplayParts(attribute.Value, state.Position, state.SemanticModel, state.Format));
         }
 
-        private static void AppendTextFromSeeTag(FormatterState state, XElement element, Compilation compilation)
+        internal static IEnumerable<SymbolDisplayPart> CrefToSymbolDisplayParts(string crefValue, int position, SemanticModel semanticModel, SymbolDisplayFormat format = null)
         {
-            var crefAttribute = element.Attribute("cref");
-
-            if (crefAttribute == null)
+            // first try to parse the symbol
+            if (semanticModel?.Compilation != null)
```

Comment:
How would either semanticModel be null, or worse, the compilation of a semantic model?